deepspeed/runtime/pipe/engine.py

```diff
@@ -241,11 +241,10 @@ def _exec_reduce_grads(self):
         self._force_grad_boundary = True
         if self.pipeline_enable_backward_allreduce:
             if self.bfloat16_enabled():
-                if self.zero_optimization_stage() == 0:
+                if self.zero_optimization_stage() < 2:
```

Comment:
Replace constant 2 with `ZeroStageEnum.gradients` as in  https://github.com/microsoft/DeepSpeed/blob/f7d71ec15376bc8188e5e8eca9c79177536c6e5d/deepspeed/runtime/zero/config.py#L67